src/libasr/pass/intrinsic_array_function_registry.h

```diff
@@ -4207,6 +4216,15 @@ namespace Parity {
         int array_rank = extract_dimensions_from_ttype(ASRUtils::expr_type(args[0]), array_dims);
 
         ASR::ttype_t* mask_type = ASRUtils::expr_type(mask);
+
+        if (!ASRUtils::is_logical(*mask_type)) {
+            diag.add(diag::Diagnostic("MASK argument to `PARITY` must be LOGICAL, but got " +
+                ASRUtils::type_to_str_with_type(mask_type),
```

Comment:
```suggestion
                ASRUtils::type_to_str_fortran(mask_type),
```
I would suggest just use this.